discovery-provider/src/tasks/prune_plays.py

```diff
@@ -0,0 +1,127 @@
+import logging
+import time
+from datetime import datetime
+
+from sqlalchemy import text
+from src.tasks.celery_app import celery
+
+logger = logging.getLogger(__name__)
+
+PLAYS_ARCHIVE_TABLE_NAME = "plays_archive"
+
+# ARCHIVE_PLAYS_QUERY
+# in a sliding window, delete the oldest plays before the cutoff date
+# delete no more than MAX_BATCH
+ARCHIVE_PLAYS_QUERY = """
+    INSERT INTO
+        plays_archive
+    SELECT
+        id,
+        user_id,
+        source,
+        play_item_id,
+        created_at,
+        updated_at,
+        slot,
+        signature,
+        :current_timestamp
+    FROM
+        plays
+    WHERE
+        plays.created_at <= :cutoff_timestamp
+    ORDER BY
+        id ASC
+    LIMIT
+        :max_batch
+    """
+
+# PRUNE_PLAYS_QUERY
+# prune those same plays that were archived
+PRUNE_PLAYS_QUERY = """
+    DELETE FROM
+        plays
+    WHERE
+        id IN (
+            SELECT
+                id
+            FROM
+                plays
+            WHERE
+                plays.created_at <= :cutoff_timestamp
+            ORDER BY
+                id ASC
+            LIMIT
+                :max_batch
+        )
+    """
+
+# Prune up to the cutoff date
+# start at all plays before 2020 (328751 total)
+# TODO move to a sliding window
+DEFAULT_CUTOFF_TIMESTAMP = datetime.fromisoformat("2020-01-01")
+
+# max number of plays to prune per run
+DEFAULT_MAX_BATCH = 80000
```

Comment:
historically this is more than sufficient but recently we're probably doing more than 240k per day, might be nice to have a buffer to let things scale up a bit. what do you think about doubling this?

Reply:
Yeah at first it won't matter because this date cutoff is static but when it's a sliding window we will have to prune at least more than we index. With the new schedule we should be at 50000 max * 8 runs a day = 400000 plays per day.